crates/cli/src/native_query/aggregation_expression.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::BTreeMap;
+use std::iter::once;
+
+use configuration::schema::{ObjectField, ObjectType, Type};
+use itertools::Itertools as _;
+use mongodb::bson::{Bson, Document};
+use mongodb_support::BsonScalarType;
+
+use super::helpers::nested_field_type;
+use super::pipeline_type_context::PipelineTypeContext;
+
+use super::error::{Error, Result};
+use super::reference_shorthand::{parse_reference_shorthand, Reference};
+
+pub fn infer_type_from_aggregation_expression(
+    context: &mut PipelineTypeContext<'_>,
+    desired_object_type_name: &str,
+    bson: Bson,
+) -> Result<Type> {
+    let t = match bson {
+        Bson::Double(_) => Type::Scalar(BsonScalarType::Double),
+        Bson::String(string) => infer_type_from_reference_shorthand(context, &string)?,
+        Bson::Array(_) => todo!("array type"),
+        Bson::Document(doc) => {
+            infer_type_from_aggregation_expression_document(context, desired_object_type_name, doc)?
+        }
+        Bson::Boolean(_) => todo!(),
+        Bson::Null => todo!(),
+        Bson::RegularExpression(_) => todo!(),
+        Bson::JavaScriptCode(_) => todo!(),
+        Bson::JavaScriptCodeWithScope(_) => todo!(),
+        Bson::Int32(_) => todo!(),
+        Bson::Int64(_) => todo!(),
+        Bson::Timestamp(_) => todo!(),
+        Bson::Binary(_) => todo!(),
+        Bson::ObjectId(_) => todo!(),
+        Bson::DateTime(_) => todo!(),
+        Bson::Symbol(_) => todo!(),
+        Bson::Decimal128(_) => todo!(),
+        Bson::Undefined => todo!(),
+        Bson::MaxKey => todo!(),
+        Bson::MinKey => todo!(),
+        Bson::DbPointer(_) => todo!(),
```

Comment:
Curious why the other number types are still `todo!` other then `Double`?  Aren't the other number types pretty easy to add here?

Reply:
Yes, it very easy to add implementations for the other scalar types. It just hasn't made it to the top of my todo list.